oscar_odin/utils.py

```diff
@@ -146,3 +146,28 @@ def get_mapped_fields(mapping, *from_field_names):
         return reduce(or_, itemgetter(*from_field_names)(keyed_mapping))
 
     return reduce(or_, keyed_mapping.values())
+
+
+def get_category_ancestors():
+    """
+    Get a mapping of all child categories with all of its ancestor categories.
+    {child_id: [ancestor1_id, ancestor2_id]}}
+    """
+    query = """
+        SELECT 
+            child.id AS child_id,
+            parent.id AS up_id
+        FROM catalogue_category AS child
+        JOIN catalogue_category AS parent
```

Comment:
I think you can do this in one join too;

```sql
        JOIN catalogue_category AS parent
        ON child.path LIKE (parent.path || '%')
            AND child.depth > parent.depth
```

Would probably be a bit more performant too.